src/Http/Controllers/VatController.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Rapidez\VatValidation\Http\Controllers;
+
+use Ibericode\Vat\Validator;
+use Ibericode\Vat\Vies\ViesException;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Cache;
+
+class VatController
+{
+    /** @return array<string, mixed> */
+    public function __invoke(Request $request): array
+    {
+        $request->validate([
+            'id' => 'string|required',
+        ]);
+
+        if (!config('rapidez.vatvalidation.allow_anywhere')) {
+            // Check if we should validate here
+            abort_unless($request->bearerToken() || $request->headers->get('referer') === route('account.register'), 401);
```

Comment:
The referrer can be spoofed, allowing others to abuse this endpoint.
We should add csrf validation here

Reply:
You can also abuse the endpoint by just adding one product to your cart, so I don't know how much value it really adds to also check csrf here 🤔